internal/clusterconfig/clusterconfig.go

```diff
@@ -36,56 +38,81 @@ const (
 	idmsFIlePath       = "image-digest-mirror-set.json"
 )
 
+var (
+	machineConfigNames = []string{
+		"99-master-ssh",
+		"99-worker-ssh",
+	}
+)
+
+// clusterConfiguration contains the current cluster's objects to be applied post-pivot to the
+// upgraded cluster.
+type clusterConfiguration struct {
+	clusterInfo    clusterinfo.ClusterInfo
+	clusterID      v1.ClusterID
+	idmsList       v1.ImageDigestMirrorSetList
+	proxy          v1.Proxy
+	pullSecret     corev1.Secret
+	machineConfigs []mcv1.MachineConfig
+}
+
 // UpgradeClusterConfigGather Gather ClusterConfig attributes from the kube-api
 type UpgradeClusterConfigGather struct {
 	client.Client
 	Log    logr.Logger
 	Scheme *runtime.Scheme
 }
 
-// FetchClusterConfig collect the current cluster config and write it as json files into data dir:
+// FetchClusterConfig collects the current cluster's configuration and write it as JSON files into
+// given filesystem directory.
 func (r *UpgradeClusterConfigGather) FetchClusterConfig(ctx context.Context, ostreeDir string) error {
 	// TODO: Add the following
-	// ssh keys
 	// Other Machine configs?
 	// additionalTrustedCA (from the images.config)
 	// Other Certificates?
 	// Catalog source
 	// ACM registration?
 	r.Log.Info("Fetching cluster configuration")
-	pullSecret := &corev1.Secret{}
-	if err := r.Client.Get(ctx, types.NamespacedName{Name: pullSecretName, Namespace: configNamespace}, pullSecret); err != nil {
+
+	clusterConfig := &clusterConfiguration{}
+
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: pullSecretName, Namespace: configNamespace}, &clusterConfig.pullSecret); err != nil {
 		return err
 	}
+
 	clusterVersion := &v1.ClusterVersion{}
 	if err := r.Client.Get(ctx, types.NamespacedName{Name: "version"}, clusterVersion); err != nil {
 		return err
 	}
-	clusterID := clusterVersion.Spec.ClusterID
-	proxy := &v1.Proxy{}
-	if err := r.Client.Get(ctx, types.NamespacedName{Name: proxyName}, proxy); err != nil {
+	clusterConfig.clusterID = clusterVersion.Spec.ClusterID
+
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: proxyName}, &clusterConfig.proxy); err != nil {
 		return err
 	}
 
+	for _, machineConfigName := range machineConfigNames {
```

Comment:
Maybe we move it to function? I really hope to split this function to smaller ones and make each resource to be handled totally independent? WDYT?

Reply:
Absolutely, I think it makes sense to have independent functions for each resource, especially if we're going to add even more resources/config here. I'll try something and let's see if it's what you also had in mind, I'll update you here.

Reply:
I just pushed an updated version of this PR with the refactoring you also suggested. WDYT? Is that close to what you had in mind?